src/gui/topbar_subwindows/military_subwindows/gui_build_unit_large_window.hpp

```diff
@@ -227,318 +165,54 @@ class build_units_listbox : public listbox_element_base<buildable_units, buildab
-												row_contents.push_back(information);
-											}
-										}
-									} else {
-										for(int32_t i = 0; i < total; i++) {
-											buildable_unit_entry_info information;
-											information.is_navy = false;
-											information.pop_info = fat_id2.get_pop().id;
-											information.province_info = fat_id2.get_province().id;
-											row_contents.push_back(information);
-										}
-									}
-								//Checks if pop is greater in size than the pop_min_size_for_regiment (1000 in vanilla)
-								//If so, number of possible regiments from pop is ONE
-								} else if(fat_id2.get_pop().get_size() >= state.defines.pop_min_size_for_regiment) {
-									int32_t total = int32_t(1);
-									for(const auto fat_id3 : fat_id2.get_pop().get_regiment_source()) {
-										total--;
-									}
-									for(const auto fat_id3 : state.world.pop_get_province_land_construction(fat_id2.get_pop())) {
-										total--;
-									}
-									auto culture_id = state.world.pop_get_culture(fat_id2.get_pop().id);
-									if(state.military_definitions.unit_base_definitions[unit_type].primary_culture) {
-										if(culture_id != state.world.nation_get_primary_culture(state.local_player_nation)) {
-											for(auto element : state.world.nation_get_accepted_cultures(state.local_player_nation)) {
-												if(culture_id.id == element) {
-													for(int32_t i = 0; i < total; i++) {
-														buildable_unit_entry_info information;
-														information.is_navy = false;
-														information.pop_info = fat_id2.get_pop().id;
-														information.province_info = fat_id2.get_province().id;
-														row_contents.push_back(information);
-													}
-												}
-											}
-										} else {
-											for(int32_t i = 0; i < total; i++) {
-												buildable_unit_entry_info information;
-												information.is_navy = false;
-												information.pop_info = fat_id2.get_pop().id;
-												information.province_info = fat_id2.get_province().id;
-												row_contents.push_back(information);
-											}
-										}
-									} else {
-										for(int32_t i = 0; i < total; i++) {
-											buildable_unit_entry_info information;
-											information.is_navy = false;
-											information.pop_info = fat_id2.get_pop().id;
-											information.province_info = fat_id2.get_province().id;
-											row_contents.push_back(information);
-										}
+		if(parent) {
+			Cyto::Any payload = dcon::unit_type_id{};
+			parent->impl_get(state, payload);
+			dcon::unit_type_id utid = Cyto::any_cast<dcon::unit_type_id>(payload);
+			if(is_navy == false) {
+				for(auto po : state.world.nation_get_province_ownership_as_nation(state.local_player_nation)) {
+					auto p = po.get_province();
+					state.world.for_each_culture([&](dcon::culture_id c) {
+						if(command::can_start_land_unit_construction(state, state.local_player_nation, p, c, utid)) {
```

Comment:
Does this lag the window? You might try filtering by cultures that are actually present (via looking in the demographics) ...

Reply:
Doesn't lag window, even on debug

Reply:
Well, I guess its fine then. Wildly wasteful, but fine